MAPL_Base/MAPL_EtaHybridVerticalCoordinate.F90

```diff
@@ -0,0 +1,282 @@
+#include "MAPL_Generic.h"
+#include "MAPL_ErrLog.h"
+#include "unused_dummy.H"
+
+module MAPL_EtaHybridVerticalCoordinate
+   use, intrinsic :: ISO_FORTRAN_ENV, only: REAL64, REAL32
+   use ESMF
+   use MAPL_ErrorHandlingMod
+   use MAPL_KeywordEnforcerMod
+   implicit none
+   private
+
+   public :: EtaHybridVerticalCoordinate
+
+
+   type :: EtaHybridVerticalCoordinate
+      private
+      real(kind=REAL64), allocatable :: ak(:)
+      real(kind=REAL64), allocatable :: bk(:)
+      integer :: num_levels = 0
+      real(kind=REAL64) :: ref_pressure
+   contains
+      procedure :: get_eta_r8
+      procedure :: get_eta_r4
+      procedure :: get_pressure_levels_r8
+      procedure :: get_pressure_levels_r4
+      procedure :: get_pressures_r8_3d
+      procedure :: get_pressures_r4_3d
+      generic :: get_eta =>get_eta_r8, get_eta_r4
+      generic :: get_pressure_levels=>get_pressure_levels_r8,   get_pressure_levels_r4
+      generic :: get_pressures      =>get_pressures_r8_3d,get_pressures_r4_3d
+   end type EtaHybridVerticalCoordinate
+
+   interface newEtaHybridVerticalCoordinate
+      module procedure new_EtaHybridVerticalCoordinate_by_ak_bk
+      module procedure new_EtaHybridVerticalCoordinate_by_cfg

```

Comment:
Need one more constructor where just the filename is passed in.   Then a config file is created and the existing constructor is called.

Reply:
done